src/core/render.ts

```diff
@@ -1,57 +1,232 @@
+import {Source} from './source/source';
 import {App as iApp} from '../internal/app';
 import {Render as iRender} from '../internal/render';
 import { exec } from '../internal/internal';
 
 export class Render {
+  private static _canvasCounter: number = 0;
+  private _index: number;
+  private _fps: number;
+  private _canvas: HTMLCanvasElement;
+  private _rendering: boolean;
+  private static _canvasPool: Render[] = [];
 
+  constructor(props?) {
+    this._index = props.canvasIndex;
+    this._fps = props.fps;
+    this._canvas = props.canvas;
+    this._rendering = props.rendering;
+  }
 
-  // Assign a canvas and create necessary webgl context
-  static setTargetCanvas(canvas: HTMLCanvasElement,
-    fps: number = 30): Promise<any> {
+  static getPreparedCanvases() {
+    return new Promise(resolve => {
+      resolve(Render._canvasPool)
+    })
+  }
+
+  /**
+   * param: (canvas: HTMLCanvasElement, fps?: number)
+   *
+   * Prepare a canvas for rendering.
+   *
+   */
+  static prepareCanvas(canvas: HTMLCanvasElement, fps: number = 30, includeToPool: boolean = true) {
+    return new Promise((resolve, reject) => {
+      const canvasProps = {
+        'canvas': canvas,
+        'fps': fps,
+        'canvasIndex': Render._canvasCounter,
+        'rendering': false
+      };
+      if (includeToPool) {
+        Render._canvasPool.push(new Render(canvasProps))
+      }
+      iRender.initializeCanvas(canvas, Render._canvasCounter, fps).then(() => {
+        resolve(true);
+      })
+      Render._canvasCounter = Render._canvasCounter + 1;
+    })
+  }
+
+  /**
+   * param: (canvas: HTMLCanvasElement)
+   *
+   * Array list of Render object of prepared canvases and their fps.
+   *
+   */
+
+  static cleanCanvas(canvas: HTMLCanvasElement) {
+    return new Promise(resolve => {
+      const match  = Render._canvasPool.filter(renObj => renObj['_canvas'] === canvas)
+      Render._canvasPool = Render._canvasPool.filter(renObj => renObj['_canvas'] !== canvas)
+      if (match.length === 1) {
+        iRender.stopCanvasToUseView(match[0]['_index']).then(res => {
+          Render.prepareCanvas(canvas, 30, false)
+          resolve(true);
+        })
+      }
+    })
+  }
+
+  /**
+   * param: (canvas: HTMLCanvasElement, state?: boolean)
+   *
+   * If there's an active render, set rendering to true or false.
+   */
+  static toggleRender(canvas: HTMLCanvasElement, state: boolean) {
+    return new Promise(resolve => {
+      const match  = Render._canvasPool.filter(renObj => renObj['_canvas'] === canvas)
+      if (match.length === 1) {
+        match[0]._rendering = state;
+        iRender.startStopRender(match[0]['_index'], state).then(res => {
+          resolve(res)
+        })
+      }
+    })
+  }
+
+    /**
+     * params: (canvas: HTMLCanvasElement, scene: any)
+     *
+     * scene parameter can be Scene instance, scene index, or scene uid.
+     *
+     */
+    static renderScene(canvas: HTMLCanvasElement, scene: any) {
```

Comment:
We could also make use of typescript's union. This is not needed, but I thought that I should share it just in case we do something similar in the future.

```suggestion
    static renderScene(canvas: HTMLCanvasElement, scene: Scene | number | string) {
```